cmd/influx_tools/parquet/exporter.go

```diff
@@ -0,0 +1,488 @@
+	for _, shard := range e.shards {
+		start := time.Now()
+		fmt.Fprintf(e.writer, "Starting export of shard %d...\n", shard.ID())
+
+		for _, measurement := range e.measurements {
+			if err := e.exportMeasurement(ctx, shard, measurement); err != nil {
+				return fmt.Errorf("exporting measurement %q in shard %d failed: %w", measurement, shard.ID(), err)
+			}
+		}
+		fmt.Fprintf(e.writer, "Finished export of shard %d in %v...\n", shard.ID(), time.Since(start))
+	}
+	fmt.Fprintf(e.writer, "Finished export in %v...\n", time.Since(e.exportStart))
+
+	return nil
+}
+
+func (e *exporter) exportMeasurement(ctx context.Context, shard *tsdb.Shard, measurement string) error {
+	startMeasurement := time.Now()
+
+	// Get the cumulative scheme with all tags and fields for the measurement
+	creator, found := e.schemata[measurement]
+	if !found {
+		return errors.New("no schema creator found")
+	}
+
+	if len(creator.fieldKeys) == 0 {
+		fmt.Fprintf(e.writer, "  Skipping measurement %q without fields\n", measurement)
+		return nil
+	}
+
+	// Create a batch processor
+	series := creator.series[shard.ID()]
+	batcher, err := newBatcher(ctx, shard, measurement, series, creator.typeConverters, creator.nameResolutions)
+	if err != nil {
+		return fmt.Errorf("creating batcher failed: %w", err)
+	}
+
+	// Check if we do have data for the measurement and skip the shard if
+	// that's not the case
+	rows, err := batcher.next(ctx)
+	if err != nil {
+		return fmt.Errorf("checking data failed: %w", err)
+	}
+	if len(rows) == 0 {
+		fmt.Fprintf(e.writer, "  Skipping measurement %q without data\n", measurement)
+		return nil
+	}
+	batcher.reset()
+
+	fmt.Fprintf(e.writer, "  Exporting measurement %q...\n", measurement)
+
+	// Create a parquet schema for writing the data
+	metadata := map[string]string{
+		"export":      e.exportStart.Format(time.RFC3339),
+		"database":    e.db,
+		"retention":   e.rp,
+		"measurement": measurement,
+		"shard":       strconv.FormatUint(shard.ID(), 10),
+		"start_time":  e.startDate.Format(time.RFC3339Nano),
+		"end_time":    e.endDate.Format(time.RFC3339Nano),
+	}
+	schema, err := creator.schema(metadata)
+	if err != nil {
+		return fmt.Errorf("creating arrow schema failed: %w", err)
+	}
+
+	// Create parquet file with a increasing sequence number per measurement
+	filename := filepath.Join(
+		e.path,
+		fmt.Sprintf("%s-%05d.parquet", measurement, e.filenames[measurement]),
+	)
+	e.filenames[measurement]++
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("creating file %q failed: %w", filename, err)
+	}
+	defer file.Close()
+
+	writer, err := pqarrow.NewFileWriter(
+		schema,
+		file,
+		parquet.NewWriterProperties(parquet.WithCreatedBy("influx_tools")),
+		pqarrow.NewArrowWriterProperties(pqarrow.WithCoerceTimestamps(arrow.Nanosecond)),
+	)
+	if err != nil {
+		return fmt.Errorf("creating parquet writer for file %q failed: %w", filename, err)
+	}
+	defer writer.Close()
```

Comment:
`Capture` error from `Close`.  A frequent source fo these errors is the disk being full.